### PR TITLE
De-RxJS-ing 

### PR DESCRIPTION
**Pull request by JoshDSommer**
removing RxJS infavor of using clearTimeout and setTimeout. cutting down on complexity and removing nested subscribes.

clearTimeout being null should be fix in zone.js as of beta.12. 

https://github.com/angular/angular/issues/7680

### CODE REVIEW

File: source/behaviors/autosave/autosave.tests.ts [hunk 1]
```diff
@@ -192,14 +192,5 @@ describe('AutosaveDirective', () => {
 			sinon.assert.calledWith(autosaveAction.trigger, waitValue);
 		});
 
-		it('should not save if the form becomes pristine immediately before saving', () => {
```
Comment (KorinaGraber) [inline source/behaviors/autosave/autosave.tests.ts:195]: We still want this. This was another thing that got corrected/added with the previous PR

File: source/behaviors/autosave/autosave.ts [hunk 2]
```diff
@@ -31,34 +28,31 @@ export class AutosaveDirective implements AfterViewInit {
...
-			this.autosaveStart$.next();
+		if (this.timer) {
+			clearTimeout(this.timer);
+			this.timer = null;
+			this.setDebounce();
 		}
 	}
 
 	autosave = (): void => {
-		if (!this.canAutosave()) {
```
Comment (KorinaGraber) [inline source/behaviors/autosave/autosave.ts:54]: We still want this. This was another thing that got corrected/added with the previous PR

**Review by KorinaGraber — CHANGES_REQUESTED** [review]
This should work. But note that I included a related fix in the previous PR, which we still want.

File: source/behaviors/autosave/autosave.ts [hunk 3]
```diff
@@ -1,5 +1,4 @@
-import { Directive, Input, Self, AfterViewInit, HostListener } from '@angular/core';
-import { Observable, Subject } from 'rxjs';
+import { Directive, Input, Self, AfterViewInit, HostListener, ElementRef } from '@angular/core';
```
Comment (KorinaGraber) [inline source/behaviors/autosave/autosave.ts]: Why the ElementRef import?

File: source/behaviors/autosave/autosave.ts (hunk 2 above)
Comment (KorinaGraber) [reply source/behaviors/autosave/autosave.ts:54]: https://github.com/RenovoSolutions/TypeScript-Angular-Components/pull/366/commits/b9cd3c11269c5d99c188e33f40405c3b5747932c

File: source/behaviors/autosave/autosave.tests.ts (hunk 1 above)
Comment (KorinaGraber) [reply source/behaviors/autosave/autosave.tests.ts:195]: https://github.com/RenovoSolutions/TypeScript-Angular-Components/pull/366/commits/b9cd3c11269c5d99c188e33f40405c3b5747932c

**Review by KorinaGraber — APPROVED** [review]

**Comment by SamGraber** [comment]
Merge conflicts

**Review by KorinaGraber — CHANGES_REQUESTED** [review]
Merge conflicts

**Review by KorinaGraber — APPROVED** [review]

**Pull request merged**